benchmarks/bench_feature_expansions.py

```diff
@@ -0,0 +1,61 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.sparse as sparse
+from sklearn.preprocessing import PolynomialFeatures
+from time import time
+
+degree = 2
+trials = 5
+num_rows = 100
+dimensionalities = np.array([50, 100, 150, 200])
```

Comment:
Maybe we could instead explore:

```python
dimensionalities = np.array([25, 50, 100, 200])
```

and then use a loglog plot so that straight lines would reflect linear time complexities and would make it easier to interpret the plots.